src/api/goods.ts

```
@@ -4,13 +4,33 @@ import { Good } from '../types/Good';
 const API_URL = `https://mate-academy.github.io/react_dynamic-list-of-goods/goods.json`;
 
 export function getAll(): Promise<Good[]> {
-  return fetch(API_URL).then(response => response.json());
+  return fetch(API_URL)
+    .then(response => {
+      if (response.ok) {
+        return response.json();
+      }
+
+      throw new Error('Failed to fetch goods');
+    })
+    .catch(error => {
+      throw error;
+    });
 }
 
 export const get5First = () => {
-  return getAll().then(goods => goods); // sort and get the first 5
+  return getAll()
+    .then(goods =>
+      goods.sort((a, b) => a.name.localeCompare(b.name)).slice(0, 5),
+    )
+    .catch(() => {
+      return [];
+    });
 };
 
-export const getRedGoods = () => {
-  return getAll().then(goods => goods); // get only red
+export const getRed = () => {
+  return getAll()
+    .then(goods => goods.filter(good => good.color === 'red'))
+    .catch(() => {
+      return [];
+    });
```

Returning an empty array in the `.catch()` block might hide potential errors. According to the checklist, you should handle errors when making a request to the server. Consider logging the error or re-throwing it to provide better error feedback.